modules/wordpress-com-block-editor-iframe.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * WordPress.com Block Editor Iframe
+ * Allow new block editor posts to be composed on WordPress.com.
+ * This is auto-loaded as of Jetpack v7.4 for sites connected to WordPress.com only.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Prevents frame options header from firing if this is a whitelisted iframe request.
+ */
+function jetpack_disable_send_frame_options_header() {
+	if ( jetpack_framing_allowed() ) {
+		remove_action( 'admin_init', 'send_frame_options_header' );
+	}
+}
+add_action( 'admin_init', 'jetpack_disable_send_frame_options_header', 9 );
+
+/**
+ * Adds custom admin body class if this is a whitelisted iframe request.
+ *
+ * @param string $classes Admin body classes.
+ * @return string
+ */
+function jetpack_add_iframed_body_class( $classes ) {
+	if ( jetpack_framing_allowed() ) {
+		$classes .= ' is-iframed ';
+	}
+
+	return $classes;
+}
+add_filter( 'admin_body_class', 'jetpack_add_iframed_body_class' );
+
+/**
+ * Checks whether this is a whitelisted iframe request.
+ *
+ * @return bool
+ */
+function jetpack_framing_allowed() {
+	if ( empty( $_GET['frame-nonce'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+		return false;
+	}
+
+	$verified = jetpack_verify_frame_nonce(
+		$_GET['frame-nonce'],  // phpcs:ignore WordPress.Security.NonceVerification
+		'frame-' . Jetpack_Options::get_option( 'id' )
+	);
+
+	if ( $verified && ! defined( 'IFRAME_REQUEST' ) ) {
+		define( 'IFRAME_REQUEST', true );
+	}
+
+	return (bool) $verified;
+}
+
+/**
+ * Verify that correct nonce was used with time limit.
+ *
+ * The user is given an amount of time to use the token, so therefore, since the
+ * UID and $action remain the same, the independent variable is the time.
+ *
+ * @param string $nonce  Nonce that was used in the form to verify.
+ * @param string $action Should give context to what is taking place and be the same when nonce was created.
+ * @return boolean Whether the nonce is valid.
+ */
+function jetpack_verify_frame_nonce( $nonce, $action ) {
+	if ( empty( $nonce ) ) {
+		return false;
+	}
+
+	list( $expiration, $user_id, $hash ) = explode( ':', $nonce, 3 );
```

Comment:
Let's check and make sure `$user_id` is not falsey here. If it is, we'll look get the blog token below (on the `Jetpack_Data::get_access_token()` line).

It shouldn't matter, but we may as well be sure.